core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -325,21 +319,19 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
       case _: LeaderElectionNotNeededException => // swallow
       case nroe: NoReplicaOnlineException => throw nroe
       case sce: Throwable =>
-        val failMsg = "encountered error while electing leader for partition %s due to: %s.".format(topicAndPartition, sce.getMessage)
-        stateChangeLogger.error("Controller %d epoch %d ".format(controllerId, controller.epoch) + failMsg)
-        throw new StateChangeFailedException(failMsg, sce)
+        val failMsg = s"Encountered error while electing leader for partition $topicAndPartition due to: ${sce.getMessage}"
+        stateChangeLog.error(failMsg)
+        throw new StateChangeFailedException(s"${stateChangeLog.logPrefix}$failMsg", sce)
```

Comment:
Guess we could also expose a `newExceptionMessage` method or something from `StateChangeLogger`. Kind of ugly either way.

Reply:
I'm replacing `logPrefix` with `messageWithPrefix`. Does that work?